### PR TITLE
[BugFix] Add SM120 fallback for DeepSeek sparse indexer logits

### PR DESCRIPTION
**Pull request by ACEEE-1222**
## Purpose

DeepSeek sparse MLA on SM120 already selects the dedicated FlashInfer attention backend, but the sparse indexer still calls DeepGEMM for MQA logits and paged scheduler metadata. Those DeepGEMM paths reject SM120 with an unsupported-architecture error.

This PR:

- routes FP8 prefill and paged-decode indexer logits through a Torch/BF16 compatibility implementation on SM120;
- avoids generating DeepGEMM scheduler metadata that the SM120 fallback does not consume;
- fails explicitly for FP4 Q, which this compatibility path does not support.

This is a correctness fallback for the indexer stage. The optimized SM120 sparse-attention stage remains handled by FlashInfer.

Related to #47266 and #47818.

## Test Plan

```bash
uvx ruff format --check   vllm/model_executor/layers/sparse_attn_indexer.py   vllm/model_executor/layers/sparse_indexer_sm120.py   vllm/v1/attention/backends/mla/indexer.py   tests/kernels/test_sparse_attn_indexer_sm120.py

uvx ruff check   vllm/model_executor/layers/sparse_attn_indexer.py   vllm/model_executor/layers/sparse_indexer_sm120.py   vllm/v1/attention/backends/mla/indexer.py   tests/kernels/test_sparse_attn_indexer_sm120.py

pytest -q tests/kernels/test_sparse_attn_indexer_sm120.py
```

Manual DeepSeek-V4-Flash TP=8 serving smoke test on 8x NVIDIA RTX 6000D (SM120), including a short chat request and a 12K-token prompt.

## Test Result

- Ruff format and lint checks passed.
- GPU unit tests: `2 passed in 6.64s`.
- Service health check returned HTTP 200.
- Short chat: HTTP 200, 13 prompt tokens + 32 completion tokens, coherent output.
- Long-context smoke: HTTP 200, 12,006 prompt tokens + 2 completion tokens.
- No Traceback, OOM, or unsupported-architecture errors in the service log.

The end-to-end smoke used the exact PR fallback implementation mounted into the available NGC 26.06 runtime, with a thin call-signature adapter for that image's older indexer API.

No documentation update is needed because this change does not alter the public API or backend-selection behavior.

---
<details>
<summary>Essential elements checklist</summary>

- [x] Purpose and related issues
- [x] Test commands
- [x] Unit and end-to-end test results
- [x] Documentation impact considered

</details>

### CODE REVIEW

**Comment by ACEEE-1222** [comment]
Closing as obsolete after discovering that vLLM #47304 already pins the
DeepGEMM nv-dev revision containing SM120 support.

This fallback would bypass the optimized DeepGEMM MQA-logits kernels on all
SM120 devices and does not support FP4 queries, so keeping it would introduce
a performance and feature regression.

The original failure was caused by a stale/mismatched DeepGEMM installation.
I will validate the current vendored DeepGEMM path and submit a narrower fix
if any unsupported shape remains.
